packages/eslint-plugin-ckeditor5-rules/tests/rules/prevent-license-key-leak.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const RuleTester = require( 'eslint' ).RuleTester;
+
+const ruleTester = new RuleTester( {
+	parser: require.resolve( '@typescript-eslint/parser' )
+} );
+
+ruleTester.run( 'prevent-license-key-leak', require( '../../lib/rules/prevent-license-key-leak' ), {
+	valid: [
```

Comment:
If we want to be _really_ safe, we should also verify:

* `get licenseKey() { ... }` - it's a getter and could keep a license key,
* `{ someKey: { someValue: { licenseKey: ... } }` - a nested value, I believe it should be ignored.

Reply:
My idea for this rule was to only detect the most simple case, when someone forgot to remove the license key after testing. It's not meant to prevent any possible case. There are other examples where this rule can be worked around, but I don't think we need to worry about them. Example:

```js
const LICENSE_KEY = '...';

const config = {
  licenseKey: LICENSE_KEY,
};
```